src/Lykke.Service.KycReports.AzureRepositories/Reports/ReportsRepository.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using AzureStorage;
+using Lykke.Service.KycReports.Core.Domain.Reports;
+
+using Newtonsoft.Json;
+
+namespace Lykke.Service.KycReports.AzureRepositories.Reports
+{
+    public class ReportsRepository : IKycReportsRepository
+    {
+        private readonly INoSQLTableStorage<ReportRowEntity> _tableStorage;
+
+        public ReportsRepository(INoSQLTableStorage<ReportRowEntity> tableStorage)
+        {
+            _tableStorage = tableStorage;
+        } 
+
+        public async Task InsertRow(IKycReportDailyLeadership row)
+        {
+            var reportDataRow = ReportRowEntity.Create(KycReportType.KycReportDailyLeadership, row, row.ReportDay.Ticks.ToString());
+            await _tableStorage.InsertOrReplaceAsync(reportDataRow);
+        }
+
+        public async Task InsertRow(IKycOfficerStatsDataReport row)
+        {
+            var reportDataRow = KycOfficerStatsReportEntity.Create(row, row.RowId);
+            await _tableStorage.InsertOrReplaceAsync(reportDataRow);
+        }
+
+        public async Task InsertRow(IKycOfficersPerformanceRow row)
+        {
+            var reportDataRow = KycOfficersPerformanceReportEntity.Create(row, row.RowId);
+            await _tableStorage.InsertOrReplaceAsync(reportDataRow);
+        }
+
+        public async Task<IEnumerable<string>> GetJsonData(KycReportType reportType, bool isDescending = false, IEnumerable<string> rowKeys = null)
+        {
+            string partitionKey;
+
+            switch (reportType)
+            {
+                case KycReportType.KycReportDailyLeadership:
+                    partitionKey = ReportRowEntity.GeneratePartitionKey(KycReportType.KycReportDailyLeadership);
+                    break;
+                default:
+                    partitionKey = null;
+                    break;
+            }
+
+
+            if (!string.IsNullOrWhiteSpace(partitionKey))
+            {
+                IEnumerable<ReportRowEntity> data;
+                if (rowKeys == null)
+                    data = await _tableStorage.GetDataAsync(partitionKey);
+                else
+                    data = await _tableStorage.GetDataAsync(partitionKey, rowKeys);
+
+                var jsonRows = isDescending
+                    ? data.OrderByDescending(d => d.Timestamp).Select(d => d.JsonRow)
+                    : data.Select(d => d.JsonRow);
+                
+                //var jsonReport = $"[\r\n{string.Join(", \r\n", jsonRows)}\r\n]";
+
+                return jsonRows;
+            }
+
+            return new List<string>();
```

Comment:
Why do we return empty list in case of empty partition key? We can do full scan for not empty list of RowKeys

Reply:
I don't think we should change such behavior at the present time